### PR TITLE
added footer and top nav links test

### LINKED ISSUE #4626: Split page links test into nav and footer links test
**Is your feature request related to a problem? Please describe.**
As part of this issue https://github.com/bbc/simorgh/issues/2228 tests to verify all the links on a page was added initially for articles pages which as a part of refactoring has been made to run on frontpage page as well which results in cypress test verifying all the story pages on the frontpage.

We have a separate test to assert whether article page on frontpage loads the page and back navigation to the frontpage works as expected.

So this `Page links test` should be separated into two tests one for the navigation links and a separate one for the footer links.

**Describe the solution you'd like**
Write tests one for asserting the navigation links and a separate one for the footer links and remove the test which asserts all the links on a page.

**Describe alternatives you've considered**

**Testing notes**
To run all the tests irrespective of the smoke value use this command locally `CYPRESS_APP_ENV=test CYPRESS_SMOKE=false npm run test:e2e:interactive`

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by PriyaKR**
Resolves #4626 
**Overall change:** _Page links test now being split into top nav links test and footer links test(wouldnt be checking each story on a frontpage)_

**Code changes:**

- _Page link test split into top nav and footer links test_
- _Also the link assertion test would output the links which 404s to make debugging easier_

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [ ] This PR requires manual testing

### CODE REVIEW

**Comment by PriyaKR** [comment]
I have separated out the page link tests to be more specific like Nav links and footer link tests.
Also as a part of this PR when a link 404s while running on jenkins CI having which link has failed in the `stdout` would make debugging quicker ie)know which link on the page is throwing errors.

But unfortunately cypress doesn't support to see console.log statements when running headlessly.
See https://github.com/cypress-io/cypress/issues/448

There are cypress plugins available to use now until cypress fixes them but those plugins seem to be available for chrome,Chromium, or Canary browsers but not electron https://github.com/flotwig/cypress-log-to-output

**Comment by jamesdonoh** [comment]
Regarding knowing which link is failing, I think we can just pass [a second argument to the `equal` Chai assertion](https://www.chaijs.com/api/bdd/#method_equal) to get the failing URL logged to the console without needing any extra `cy.log` or a plugin, for example:
```javascript
expect(resp.status).to.not.equal(404, `Received 404 for ${url}`);
```

**Review by ryanmccombe — APPROVED** [review]

**Review by oluoluoxenfree — APPROVED** [review]

File: cypress/integration/pages/testsForAllPages.js [hunk 1]
```diff
@@ -382,18 +382,28 @@ export const testsThatNeverRunDuringSmokeTestingForAllPageTypes = ({
...
+            cy.request({
+              url,
+              failOnStatusCode: false,
+            }).then(resp => {
+              expect(resp.status).to.not.equal(404, `Received 404 for ${url}`);
             });
+          });
+        });
+
+        it('Footer links should not 404', () => {
```
Comment (jamesdonoh) [inline cypress/integration/pages/testsForAllPages.js:397]: Please could we make the test description consistent between here and L385? As they are basically doing the same thing but for different page regions.

**Review by jamesdonoh — CHANGES_REQUESTED** [review]

File: cypress/integration/pages/testsForAllPages.js (hunk 1 above)
Comment (PriyaKR) [reply cypress/integration/pages/testsForAllPages.js:397]: Done.

**Review by jamesdonoh — APPROVED** [review]
Thanks, looks good 👍

**Pull request merged**